lib/records/RecCore.cc

```diff
@@ -717,7 +727,8 @@ RecGetRecordDefaultDataString_Xmalloc(char *name, char **buf, bool lock)
 
   if (auto it = g_records_ht.find(name); it != g_records_ht.end()) {
     RecRecord *r = it->second;
-    *buf         = (char *)ats_malloc(sizeof(char) * 1024);
```

Comment:
This is where it really helps, making sure that indentation helps understanding the semantics (these are two different type of assignments, local scope vs not)